### PR TITLE
Handle breaking changes in package:vm_service v10

### PR DESCRIPTION
**Pull request by derekxu16**
The `InboundReference` and `RetainingObject` types will be changed in `package:vm_service` version 10.0.0 (see [this CL](https://dart-review.googlesource.com/c/sdk/+/268521/14/runtime/vm/service/service.md)). This handles those changes. The also handles the optional parameters added to `getInstances` in `package:vm_service` version 10.1.0.

### CODE REVIEW

File: packages/devtools_app/test/vm_developer/vm_developer_test_utils.dart [hunk 1]
```diff
@@ -96,6 +96,14 @@ final testRetainingObjects = [
     parentListIndex: 1,
     parentField: 'fooParentField',
```
Comment (derekxu16) [inline packages/devtools_app/test/vm_developer/vm_developer_test_utils.dart:97]: This test is used to check which property takes priority between `parentListIndex` and `parentField`, but it should be impossible for both of these properties ot appear on the same `RetainingObject`. Do we want to change this?

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart [hunk 2]
```diff
@@ -553,13 +556,12 @@ class _RetainingObjectDescription extends StatelessWidget {
         ).buildTextSpan(context),
       );
     } else {
-      entries.addAll([
-        const TextSpan(text: 'of '),
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:557]: Why was this removed?

File: packages/devtools_app/pubspec.yaml [hunk 3]
```diff
@@ -57,7 +57,7 @@ dependencies:
   string_scanner: ^1.1.0
   url_launcher: ^6.1.0
   url_launcher_web: ^2.0.6
-  vm_service: ^9.0.0
+  vm_service: '>=10.1.0'
```
Comment (bkonyi) [inline packages/devtools_app/pubspec.yaml:60]: This can be `^10.1.0`

File: packages/devtools_app/test/vm_developer/vm_developer_test_utils.dart (hunk 1 above)
Comment (bkonyi) [reply packages/devtools_app/test/vm_developer/vm_developer_test_utils.dart:113]: Yeah, if it doesn't make sense anymore, we should update the test.

File: packages/devtools_test/pubspec.yaml [hunk 4]
```diff
@@ -27,7 +27,7 @@ dependencies:
   mockito: ^5.0.9
   path: ^1.8.0
   provider: ^6.0.2
-  vm_service: ^9.0.0
+  vm_service: '^10.1.0'
```
Comment (bkonyi) [inline packages/devtools_test/pubspec.yaml:30]: These quotations aren't necessary.

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart (hunk 2 above)
Comment (derekxu16) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart:557]: It creates two "of"s in a row, because these lines `entries` always ends with "of" after these lines https://github.com/flutter/devtools/pull/4846/files#diff-8483d57bb25f0c712d2792fc77a3f949a5f9be04fbb779e1e0e966f3a9fd3a64R523-R528

**Review by bkonyi — APPROVED** [review]
